docs/architecture/blazor-for-web-forms-developers/state-management.md

```diff
@@ -1,23 +1,89 @@
+
+## Maintain state with Session
+
+Web Forms developers could maintain information about the currently acting user with the <xref:Microsoft.AspNetCore.Http.ISession?displayProperty=nameWithType> dictionary object. It's easy enough to add an object with a string key to the `Session`, and that object would be available at a later time during the user's interactions with the application. In an attempt to eliminate managing interacting with HTTP, the `Session` object made it easy to maintain state.
+
+Session is available in ASP.NET Core and Blazor Server, but is discouraged from use in favor of storing data in a data repository appropriately. Session state is also not functional when GDPR support is activated and your visitors decline HTTP cookies in your application.
+
+Configuration for ASP.NET Core and Session state is available in the [Session and state management in ASP.NET Core article](/aspnet/core/fundamentals/app-state).
+
+## Application state
+
+The `Application` object in the Web Forms framework provides a massive, cross-request repository for interacting with application-scope configuration and state. Application state was an ideal place to store various application configuration properties that would be referenced by all requests, regardless of the user making the request. The problem with the `Application` object was that data didn't persist across multiple servers. The state of the application object was lost between restarts.
+
+As with `Session`, it's recommended that data move to a persistent backing store that could be accessed by multiple server instances. If there is volatile data that you would like to be able to access across requests and users, you could easily store it in a static property or in a singleton object that can be injected into components that require this information or interaction.
+
+The construction of an object to maintain application state and its consumption could resemble the following implementation:
+
+```csharp
+public class MyApplicationState
+{
+    public int VisitorCounter { get; private set; } = 0;
+
+    public void IncrementCounter() => VisitorCounter += 1;
+}
+```
+
+```csharp
+app.AddSingleton<MyApplicationState>();
+```
+
+```razor
+@inject MyApplicationState AppState
+
+<label>Total Visitors: @AppState.VisitorCounter</label>
+```
+
+The `MyApplicationState` object is maintained only once on the server, and the value `VisitorCounter` is fetched and output in the component's label. The `VisitorCounter` value should be persisted and retrieved from a backing data store for durability and scalability.
+
+## In the browser
+
+With single-page application (SPA) architecture, developers grew accustomed to storing data in variables or in page state. The data would be available later in the application. There are two browser features that allow for persistence of data in different scopes of the user's browser:
```

Comment:
```suggestion
Application data can also be stored client-side on the user's device so that is available later. There are two browser features that allow for persistence of data in different scopes of the user's browser:
```